spec/lang/well-formed.md

```diff
@@ -54,12 +75,6 @@ impl Type {
     fn check_wf<T: Target>(self) -> Result<()> {
         use Type::*;
 
-        // Ensure that the size is valid and a multiple of the alignment.
-        let size = self.size::<T>();
-        ensure_wf(T::valid_size(size), "Type: size not valid")?;
-        let align = self.align::<T>();
-        ensure_wf(size.bytes() % align.bytes() == 0, "Type: size is not multiple of alignment")?;
```

Comment:
Why did move this to the end of the function?

Reply:
The size calculation assumes the type to be well formed, which isn't true at the beginning of the function. E.g. the array does expect_sized of the element times the count. Now this only works after the sizedness has been asserted.

Reply:
Okay, makes sense. Please add a comment explaining that.